components/pagination.vue

```diff
@@ -0,0 +1,217 @@
+<!-- Adapted from https://github.com/lokyoung/vuejs-paginate -->
+<template>
+  <ul :class="containerClass">
+    <li v-for="page in pages" :class="[pageClass, page.selected ? activeClass : '', page.disabled ? disabledClass : '']" :key="page.index">
+      <template v-if="page.breakView">
+        <input 
+          v-if="editing === page.index"
+          :class="pageLinkClass"
+          @keyup.enter.prevent="handlePageSelected(parseInt($event.target.value, 10));"
+          :max="pageCount"
+          min="1"
+          type="number"
+        />
+        <a
+          v-else
+          :class="pageLinkClass"
+          @click="editing = page.index;"
+          tabindex="0"
+        >...</a>
+      </template>
+      <a v-else-if="page.disabled" :class="pageLinkClass" tabindex="0">{{ page.content }}</a>
+      <a
+        v-else
+        @click="handlePageSelected(page.index + 1)"
+        :class="pageLinkClass"
+        tabindex="0"
+      >{{ page.content }}</a>
+    </li>
+  </ul>
+</template>
+
+<script>
+  export default {
+    props: {
+      value: {
+        type: Number,
+      },
+      pageCount: {
+        type: Number,
+        required: true,
+      },
+      clickHandler: {
+        type: Function,
+        default: () => { },
+      },
+      pageRange: {
+        type: Number,
+        default: 3,
+      },
+      marginPages: {
+        type: Number,
+        default: 1,
+      },
+      activeClass: {
+        type: String,
+        default: 'active',
+      },
+      disabledClass: {
+        type: String,
+        default: 'disabled',
+      },
+    },
+    data() {
+      return {
+        innerValue: 1,
+        containerClass: 'pages',
+        pageClass: 'page-item',
+        pageLinkClass: 'page-link-item',
+        editing: -1,
+      };
+    },
+    computed: {
+      selected: {
+        get() {
```

Comment:
These functions are not used. Do we need them?

Reply:
They are getters and setters for the property (https://www.w3schools.com/js/js_object_accessors.asp)
They are used everywhere you see `this.selected`.